apps/dsky_clock/app.js

```diff
@@ -0,0 +1,373 @@
+  AltDrawTimer = setTimeout( function() {
+    Bangle.setHRMPower(true, 'dsky_clock');
+    let hrm=getHRM();
+    setDATA('R1',hrm.bpm);
+    setDATA('R2',hrm.bpmConfidence);
+    setDATA('R3',getSteps());
+    mode_HRM();
+  }, 5000);
+  Bangle.setHRMPower(false);
+}
+
+function mode_weather() {
+  let weather=getWeather();
+  weather.temp = Math.round(weather.temp-273.15);
+  setDATA('R1',weather.temp);
+  setDATA('R2',weather.hum);
+  setDATA('R3',weather.code);
+  setWORD('NOUN',weather.hum);
+  setWORD('VERB',weather.rain);
+}
+
+function mode_compass() {
+  AltDrawTimer = setTimeout ( function() {
+    setLight('COMPACTY','',true,Light_COMPACTY); //isCompassOn seems to be incorrect?
+    Bangle.setCompassPower(1);
+    let compass=Bangle.getCompass();
+    setDATA('R1',compass.heading);
+    setDATA('R2');
+    setDATA('R3');
+    mode_compass();
+  }, 200);
+  Bangle.setCompassPower(0);
+}
+
+function mode_GPS() {
+  setLight('COMPACTY','',true,Light_COMPACTY);
+  AltDrawTimer = setTimeout( function() {
+    Bangle.setGPSPower(1,'dsky_clock');
+    let gps=Bangle.getGPSFix();
+    setWORD('NOUN',gps.fix);
+    setWORD('VERB',gps.satellites);
+    setDATA('R1',gps.lat);
+    setDATA('R2',gps.lon);
+    setDATA('R3',gps.speed);
+    mode_GPS();
+  }, 5000);
+  Bangle.setGPSPower(0);
+}
+
+function mode_accel() {
+  AltDrawTimer = setTimeout( function() {
+    setLight('COMPACTY','',isActive(),Light_COMPACTY);
+    let accel=Bangle.getAccel();
+    setDATA('R1',accel.x);
+    setDATA('R2',accel.y);
+    setDATA('R3',accel.z);
+    mode_accel();
+  }, 100);
+}
+
+//////////// Main
+
+var drawTimeout;
+// schedule a draw for the next minute
+function queueDraw() {
+  if (drawTimeout) clearTimeout(drawTimeout);
+  drawTimeout = setTimeout(function() {
+    drawTimeout = undefined;
+    mode = 0;
+    if (AltDrawTimer) clearTimeout(AltDrawTimer);
+    drawMain();
+  }, 60000 - (Date.now() % 60000));
+}
+
+Bangle.on('lock',on=>{
+  mode = 0;
+  drawMain(); // draw immediately
+  });
+Bangle.on("message",function() { setLight('COMPACTY','',isActive(),Light_COMPACTY);});
+Bangle.on('charging',drawMain);
+NRF.on('connect',function() { setLight('L3','BT',!isBTConnected(),Light_warn); });
+NRF.on('disconnect',function() { setLight('L3','BT',!isBTConnected(),Light_warn); });
+
+Bangle.on('swipe', function(directionLR) {
+  if (directionLR == 1) {
+    mode=mode-1;
+  }
+  if (directionLR == -1) {
+    mode=mode+1;
+  }
+  if (mode < 0 ) { mode=5; }
+  mode=(mode % 6);
+  drawAlt(mode);
+});
+
+g.clear();
+draw_bg();
+
+// Show launcher when middle button pressed
+Bangle.setUI({
+  mode : "clock",
+  remove : function() {
+    if (drawTimeout) clearTimeout(drawTimeout);
+    if (AltDrawTimer) clearTimeout(AltDrawTimer);
+    drawTimeout = undefined;
+    AltDrawTimer = undefined;
+    require("widget_utils").show(); // re-show widgets
+  }});
+Bangle.loadWidgets();
+require("widget_utils").swipeOn(); // hide widgets, make them visible with a swipe
```

Comment:
For some reason the widgets are not off screen for me. Don't know why since I'd expect this `swipeOn()` call to hide them.

Reply:
think this was related to the layout setUI issues too, seems to work now.